modules/vba_documents/app/controllers/vba_documents/application_controller.rb

```diff
@@ -8,9 +8,9 @@ class ApplicationController < ::ApplicationController
     before_action :require_gateway_origin
 
     def require_gateway_origin
-      if Rails.env.production? && params[:source].blank? && Flipper.enabled?(:benefits_require_gateway_origin)
-        raise Common::Exceptions::Unauthorized
-      end
+      raise Common::Exceptions::Unauthorized if Rails.env.production? \
+        && (request.headers['X-Consumer-ID'].blank? || consumer.blank?) \
+        && Flipper.enabled?(:benefits_require_gateway_origin)
```

Comment:
Note that there is a `consumer` method which exposes `request.headers['X-Consumer-Username']` just below this excerpt